tensorflow_io/core/kernels/avro/atds/decoder_test_util.h

```diff
@@ -0,0 +1,404 @@
+
+  string schema_;
+  size_t num_of_features_;
+};
+
+template <typename T>
+DataType GetDataType() {
+  return DataTypeToEnum<T>().value;
+}
+
+template <>
+inline DataType GetDataType<string>() {
+  return DT_STRING;
+}
+
+inline std::vector<uint8_t> StringToByte(const std::string& s) {
+  std::vector<uint8_t> result;
+  result.reserve(s.size());
+  std::copy(s.begin(), s.end(), std::back_inserter(result));
+  return result;
+}
+
+inline std::string ByteToString(const std::vector<uint8_t>& t) {
+  std::string result;
+  std::copy(t.begin(), t.end(), std::back_inserter(result));
+  return result;
+}
+
+// avro::Type is used to differentiate between byte and string, both of which
+// map to datatype
+template <typename T>
+void AddDenseValue(avro::GenericDatum& datum, const string& name,
+                   const T& value) {
+  auto& record = datum.value<avro::GenericRecord>();
+  auto& feature = record.field(name);
+  feature.value<T>() = value;
+}
+
+template <typename T>
+void AddDenseValue(avro::GenericDatum& datum, const string& name,
+                   const std::vector<T>& values) {
+  auto& record = datum.value<avro::GenericRecord>();
+  auto& feature = record.field(name).value<avro::GenericArray>();
+  auto& feature_values = feature.value();
+  for (T value : values) {
+    feature_values.emplace_back(value);
+  }
+}
+
+template <>
+inline void AddDenseValue(avro::GenericDatum& datum, const string& name,
+                          const byte_array& value) {
+  auto& record = datum.value<avro::GenericRecord>();
+  auto& feature = record.field(name);
+  feature.value<byte_array>() = value;
+}
+
+template <typename T>
+inline void AddDenseValue(avro::GenericDatum& datum, const string& name,
+                          const std::vector<std::vector<T>>& values) {
+  auto& record = datum.value<avro::GenericRecord>();
+  auto& feature = record.field(name).value<avro::GenericArray>();
+  auto& sub_array_schema = feature.schema()->leafAt(0);
+
+  auto& feature_values = feature.value();
+  for (size_t i = 0; i < values.size(); i++) {
+    feature_values.emplace_back(sub_array_schema);
+    auto& sub_array = feature_values.back().value<avro::GenericArray>().value();
+    for (size_t j = 0; j < values[i].size(); j++) {
+      sub_array.emplace_back(values[i][j]);
+    }
+  }
+}
+
+template <>
+inline void AddDenseValue(avro::GenericDatum& datum, const string& name,
+                          const std::vector<byte_array>& values) {
+  auto& record = datum.value<avro::GenericRecord>();
+  auto& feature = record.field(name).value<avro::GenericArray>();
+  auto& feature_values = feature.value();
+  for (byte_array value : values) {
+    feature_values.emplace_back(value);
+  }
+}
+
+template <typename T>
+void AddSparseValue(avro::GenericDatum& datum, const string& name,
+                    const std::vector<std::vector<long>>& indices,
+                    const std::vector<T>& values) {
```

Comment:
@lijuanzhang78 there is a build error on mac that is caused due to this template function:

```bash
2023-04-22T18:39:56.1638890Z tensorflow_io/core/kernels/avro/atds/atds_decoder_test.cc:76:3: error: no matching function for call to 'AddSparseValue'
2023-04-22T18:39:56.1639280Z   AddSparseValue(atds_datum, feature_names[3], sparse_int_1d_indices,
2023-04-22T18:39:56.1639530Z   ^~~~~~~~~~~~~~
2023-04-22T18:39:56.1640170Z ./tensorflow_io/core/kernels/avro/atds/decoder_test_util.h:147:6: note: candidate function template not viable: no known conversion from 'vector<vector<int64_t>>' to 'const vector<vector<long>>' for 3rd argument
2023-04-22T18:39:56.1640640Z void AddSparseValue(avro::GenericDatum& datum, const string& name,
2023-04-22T18:39:56.1640880Z      ^
```

Can you modify the test case to handle this issue?

Reply:
@kvignesh1420 Hi! The mac build should be good now.